test/spec/SpecHelper.hs

```diff
@@ -243,3 +244,20 @@ isErrorFormat s =
   obj = decode s :: Maybe (M.Map Text Value)
   keys = maybe S.empty M.keysSet obj
   validKeys = S.fromList ["message", "details", "hint", "code"]
+
+-- | Follows these steps to verify if the table data changed in the db:
+--  * Verifies the table data in the db before the change
+--  * Does the mutation
+--  * Verifies that the table data changed in the db
+--  * Resets the table with the original data
+verifyMutation :: ByteString -> Value -> ResponseMatcher -> WaiExpectation m -> WaiSession m ()
+verifyMutation tblName dataBefore dataAfter mutation = do
+  get ("/" <> tblName) `shouldRespondWith` [json|#{dataBefore}|]
+  mutation
+  get ("/" <> tblName <> "?order=id") `shouldRespondWith` dataAfter
```

Comment:
`?order=id` being hidden is a bit surprising. How about specifying this in each request instead?